src/Microsoft.Extensions.SecretManager.Tools/Internal/MsBuildProjectReader.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ProjectModel.Internal;
+using GracefulException = Microsoft.Extensions.SecretManager.Tools.Internal.GracefulException;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    internal class MsBuildProjectReader
+    {
+        public string ReadUserSecretId(IFileInfo fileInfo)
+        {
+            if (fileInfo == null || !fileInfo.Exists)
+            {
+                throw new GracefulException($"Could not find file '{fileInfo.PhysicalPath}'");
+            }
+
+            var projectDir = Path.GetDirectoryName(fileInfo.PhysicalPath);
+
+            var dotNetHome = Path.GetDirectoryName(new Muxer().MuxerPath);
+            var sdkResolver = new DotNetCoreSdkResolver(dotNetHome);
+            var sdk = sdkResolver.ResolveProjectSdk(projectDir);
+
+            // workaround limitations in using MSBuild API
+            Environment.SetEnvironmentVariable("MSBUILD_EXE_PATH", Path.Combine(sdk.BasePath, "MSBuild.exe"));
```

Comment:
Okay yeah we don't have a good story yet for these project tools.  I guess this is valid but hopefully this isn't abused.

I'm also working on a change to build `MSBuild.dll` instead of `MSBuild.exe` like we should be.  This would break this tool if you try and take a newer package and I'm still investigating who else will be broken.